example.nelua

```diff
@@ -87,7 +87,7 @@ local systems_data = (@SystemsData){
 -- rightmost_system will use the locals
 -- anyway, do what you think is better, is just a function ;)
 
-local function movement_system(mov_system_data: SystemData, velocity_storage: VelocityStorage, position_storage: PositionStorage)
+local function movement_system(mov_system_data: SystemData*, velocity_storage: VelocityStorage*, position_storage: PositionStorage*)
```

Comment:
Records containing other records that should not be passed by value (such as `vector(T)`) should be passed by reference too. Otherwise any write operation on the inner copied vector will be invalid and cause crashes. In the future the vector may be copied to not cause crashes, but this will be an expansive operation and unwanted too or trigger a compile time error (I'm deciding yet).